third_party/gfxreconstruct/framework/decode/file_processor.cpp

```diff
@@ -2114,10 +2121,23 @@ bool FileProcessor::ProcessFrameMarker(const format::BlockHeader& block_header,
 
     if (success)
     {
-        // Validate frame end marker's frame number matches current_frame_number_ when capture_uses_frame_markers_ is
-        // true.
-        GFXRECON_ASSERT((marker_type != format::kEndMarker) || (!capture_uses_frame_markers_) ||
-                        (current_frame_number_ == (frame_number - first_frame_)));
+        // GOOGLE: [single-frame-looping] This check does not apply in the single frame looping case
```

Comment:
i am wondering if it would be cleaner to derive a class from FileProcessor, and do all our modifications in that class in a separate file to avoid modifying both file_processor.cpp/h. we only need to implement following funcs:
ProcessFrameMarker
ProcessStateMarker (only need to handle marker_type == format::kEndMarker case)